includes/class-wp-document-revisions-admin.php

```diff
@@ -884,6 +890,83 @@ public function profile_update_cb() {
 		}
 	}
 
+	/**
+	 * Ensures that any system limit on revisions does not apply to documents
+	 *
+	 * @since 3.2.2
+	 *
+	 * @param int     $num  default value for the number of revisions for the post_type
+	 * @param WP_Post $post current post
+	 */
+	public function manage_document_revisions( $num, $post ) {
+
+		if ( ! is_object( $post ) || 'document' !== $post->post_type ) {
+			return $num;
+		}
+
+		// Set default number as unlimited
+		/**
+		 * Filters the number of revisions to keep for documents.
+		 *
+		 * This should normally be unlimited and setting it can make attachments unaccessible.
+		 *
+		 * @since 3.2.2
+		 *
+		 * @param int -1 (unlimited)
+		 */
+		$num = apply_filters( 'document_max_revisions', -1 );
+
+		return $num;
+
+	}
+
+
+	/**
+	 * Ensures that an error box appears if the revisions for a post has reached a system limit
+	 *
+	 * @since 3.2.2
+	 */
+	public function check_document_revisions() {
+
+		global $post;
+
+		if ( ! is_object( $post ) || 'document' !== $post->post_type ) {
+			return;
+		}
+
+		$num = wp_revisions_to_keep( $post );
+
+		if ( 0 === $num ) {
+			// setting revisions to 0 makes no sense for this plugin
+			// @codingStandardsIgnoreLine WordPress.XSS.EscapeOutput.OutputNotEscaped
+			echo '<div class="notice notice-error"><p>';
```

Comment:
Would it make sense to `?><div class="notice notice-error"><p><?php` to avoid having to disable the linter?

Reply:
Agreed. It will remove clutter.